doc/source/whatsnew/v1.0.0.rst

```diff
@@ -102,6 +102,30 @@ String accessor methods returning integers will return a value with :class:`Int6
 We recommend explicitly using the ``string`` data type when working with strings.
 See :ref:`text.types` for more.
 
+.. _whatsnew_100.boolean:
+
+Boolean data type with missing values support
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We've added :class:`BooleanDtype` / :class:`~arrays.BooleanArray`, an extension
+type dedicated to boolean data that can hold missing values. With the default
+``'bool`` data type based on a numpy bool array, the column can only hold
```

Comment:
do we have a dedicated doc section that you can link?

Reply:
> do we have a dedicated doc section that you can link?

Going to write one if we add the actual NA behaviour to boolean array (as this is the actual different with bool dtype that needs to be explained)